packages/block-editor/src/components/rich-text/index.js

```diff
@@ -592,6 +632,20 @@ export class RichText extends Component {
 		event.preventDefault();
 	}
 
+	/**
+	 * Updates the selection state as soon as possible, event before the next
+	 * `selectionchange` event.
+	 * In order to handle subsequent events that rely on selection correctly, we
+	 * need to set the new selection as soon as we can. The `selectionchange`
+	 * event happens too late. The order is as follows: `keydown`, `paint`, then
```

Comment:
Is there any concern this specific flow of expected behavior might be browser-dependent? Worth doing a run-through in supported browsers, if not already done?

Reply:
I've tested in Firefox, but I cannot run the Puppeteer tests there (for now). This was specifically added to make it work with Puppeteer.